meetings/2021-07/july-15.md

```diff
@@ -0,0 +1,734 @@
+
+
+## `getOriginals` for stage 1
+Presenter: Jordan Harband (JHD)
+
+
+USA: Yeah. And next up we have Jordan couldn't do you want to ask for stage 1? Yeah. 
+
+JHD: So even though I don't yet have a repository, we sort of sketched it out here. I would be happy make one tomorrow after I get some sleep. It's is there any reason that could not be stage one already? I would be more comfortable with sleeping more. 
+
+SYG: So, I don't want to slow down your word, I don't want them walking, so let me hear from you. Do you think that if we wait for an official explain it, I'm proposing everything through grants X3 would that slow down velocity for you?
+
+JHD: No, it's just a nice signal. I think stage one is a problem statement, right? So the specifics of the solution don't need to be explained or understood just yet but - like I'm happy to write up that explainer, it's more like, if the problem is unclear to someone such that they are not willing to or not comfortable, making it stage one without that explainer, then obviously would need to wait. But there's a possibility, we don't meet tomorrow and so I otherwise I would just say let's revisit it tomorrow. 
+
+GC: I say given the confusion surrounding at least the discussion on this API in Matrix over the last ten minutes. think it would be good to have some. Some some things written down concretely that we can understand before discussing this more, 
+
+CP: yeah, you're on a weekend weekend, trailblaze these and get it next meeting for stage 1 and 2 maybe. I can't even get all the pieces in.
+
+WH: I'm also a bit unclear as to what is going into stage one and I would note that there is a deadline to post documents a number of days in advance of a meeting so we can review them.
+
+JHD:  Sure. And I mean that and the lack of materials are both valid, Reasons to procedurally, reject stage 1. I just but that doesn't mean you have to reject it but it sounds it sounds like they're not, there's at least a few people that aren't comfortable with stage one.
+
+WH: Yeah, but I also want to mention that I do like where you're going with this. 
+
+JWK: So I didn't catch up with the conversation before. What is the use case? Why do we need an API for it? 
+
+JHD: So that people can write code, that can't be trivially broken by someone later messing with the environment like node core uses this, because if do delete function.prototype.call node core just craps itself and dies immediately. And so there's all over nerdcore. They're trying to slowly patch this problem so that its robust against user modification of the platform and all of my npm packages are written in ways where as long as they run first. In a before user code is mess with anything, then they are robust against user code messing with things. And get intrinsic would be a much more convenient and potentially more performant although that's less of a concern for me way to to write these this robust code.
```

Comment:
```suggestion
JHD: So that people can write code, that can't be trivially broken by someone later messing with the environment like node core uses this, because if I do `delete Function.prototype.call` node core just craps itself and dies immediately. All over node core, they're trying to slowly patch this problem so that its robust against user modification of the platform and all of my npm packages are written in ways where as long as they run first, before user code messes with anything, then they are robust against user code messing with things later. getIntrinsic would be a much more convenient and potentially more performant (although that's less of a concern for me) way to to write this robust code.
```